pkg/credentials/credentials.go

```diff
@@ -41,30 +41,96 @@ var (
 			MountPath:          "/credentials-gcp",
 			EnvCredentialsFile: common.GCPCredentialsEnvKey,
 			PluginName:         common.VeleroPluginForGCP,
-			PluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_GCP_PLUGIN_REPO"), os.Getenv("VELERO_GCP_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginMicrosoftAzure: {
 			IsCloudProvider:    true,
 			SecretName:         "cloud-credentials-azure",
 			MountPath:          "/credentials-azure",
 			EnvCredentialsFile: common.AzureCredentialsFileEnvKey,
 			PluginName:         common.VeleroPluginForAzure,
-			PluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_AZURE_PLUGIN_REPO"), os.Getenv("VELERO_AZURE_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginOpenShift: {
 			IsCloudProvider: false,
 			PluginName:      common.VeleroPluginForOpenshift,
-			PluginImage:     fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_REPO"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginCSI: {
 			IsCloudProvider: false,
 			//TODO: Check if the Registry needs to an upstream one from CSI
-			PluginName:  common.VeleroPluginForCSI,
-			PluginImage: fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_CSI_PLUGIN_REPO"), os.Getenv("VELERO_CSI_PLUGIN_TAG")),
+			PluginName: common.VeleroPluginForCSI,
 		},
 	}
 )
 
+func getPluginOverrideImage(velero *oadpv1alpha1.Velero) string {
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.AWSImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.AWSImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.OpenShiftImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.OpenShiftImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.AzureImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.AzureImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.CSIImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.CSIImageKey]
+	}
+	if velero.Spec.UnsupportedOverrides[oadpv1alpha1.GCPImageKey] != "" {
+		return velero.Spec.UnsupportedOverrides[oadpv1alpha1.GCPImageKey]
+	}
+	// TODO: handle better
+	return ""
+}
+
+func getPluginImage(velero *oadpv1alpha1.Velero) string {
+	// TODO: this assumes no velero image override..handle better
+	if velero.Spec.UnsupportedOverrides != nil {
+		return getPluginOverrideImage(velero)
+	}
+	// AWS
+	for _, plugin := range velero.Spec.DefaultVeleroPlugins {
+		if plugin == oadpv1alpha1.DefaultPluginAWS {
+			if os.Getenv("VELERO_AWS_PLUGIN_REPO") == "" {
+				return "quay.io/konveyor/velero-plugin-for-aws:konveyor-oadp"
```

Comment:
These are constants in the `common` package as well, I would use those when possible